src/Module/Profile.php

```diff
@@ -0,0 +1,353 @@
+
+		if ($remote_contact) {
+			$cdata = ContactModel::getPublicAndUserContacID(remote_user(), $a->profile['profile_uid']);
+			if (!empty($cdata['user'])) {
+				$groups = Group::getIdsByContactId($cdata['user']);
+				$remote_cid = $cdata['user'];
+			}
+		}
+
+		if (!empty($a->profile['hidewall']) && !$is_owner && !$remote_contact) {
+			notice(L10n::t('Access to this profile has been restricted.') . EOL);
+			return '';
+		}
+
+		if (!$update) {
+			$tab = false;
+			if (!empty($_GET['tab'])) {
+				$tab = Strings::escapeTags(trim($_GET['tab']));
+			}
+
+			$o .= ProfileModel::getTabs($a, $is_owner, $a->profile['nickname']);
+
+			if ($tab === 'profile') {
+				$o .= ProfileModel::getAdvanced($a);
+				Hook::callAll('profile_advanced', $o);
+				return $o;
+			}
+
+			$o .= Widget::commonFriendsVisitor($a->profile['profile_uid']);
+
+			$commpage = $a->profile['page-flags'] == User::PAGE_FLAGS_COMMUNITY;
+			$commvisitor = $commpage && $remote_contact;
+
+			$a->page['aside'] .= posted_date_widget(System::baseUrl(true) . '/profile/' . $a->profile['nickname'], $a->profile['profile_uid'], true);
+			$a->page['aside'] .= Widget::categories(System::baseUrl(true) . '/profile/' . $a->profile['nickname'], (!empty($category) ? XML::escape($category) : ''));
+			$a->page['aside'] .= Widget::tagCloud();
+
+			if (Security::canWriteToUserWall($a->profile['profile_uid'])) {
+				$x = [
+					'is_owner' => $is_owner,
+					'allow_location' => ($is_owner || $commvisitor) && $a->profile['allow_location'],
+					'default_location' => $is_owner ? $a->user['default-location'] : '',
+					'nickname' => $a->profile['nickname'],
+					'lockstate' => is_array($a->user)
+					&& (strlen($a->user['allow_cid'])
+						|| strlen($a->user['allow_gid'])
+						|| strlen($a->user['deny_cid'])
+						|| strlen($a->user['deny_gid'])
+					) ? 'lock' : 'unlock',
+					'acl' => $is_owner ? ACL::getFullSelectorHTML($a->user, true) : '',
+					'bang' => '',
+					'visitor' => $is_owner || $commvisitor ? 'block' : 'none',
+					'profile_uid' => $a->profile['profile_uid'],
+				];
+
+				$o .= status_editor($a, $x);
+			}
+		}
+
+		// Get permissions SQL - if $remote_contact is true, our remote user has been pre-verified and we already have fetched his/her groups
+		$sql_extra = Item::getPermissionsSQLByUserId($a->profile['profile_uid'], $remote_contact, $groups, $remote_cid);
+		$sql_extra2 = '';
+
+		if ($update) {
+			$last_updated = (defaults($_SESSION['last_updated'], $last_updated_key, 0));
+
+			// If the page user is the owner of the page we should query for unseen
+			// items. Otherwise use a timestamp of the last succesful update request.
+			if ($is_owner || !$last_updated) {
+				$sql_extra4 = " AND `item`.`unseen`";
+			} else {
+				$gmupdate = gmdate(DateTimeFormat::MYSQL, $last_updated);
+				$sql_extra4 = " AND `item`.`received` > '" . $gmupdate . "'";
+			}
+
+			$items_stmt = DBA::p(

```

Comment:
maybe we can add it into `Item::getParentUri($profile_id)` ?

Reply:
I'll confess I didn't look too hard into what it was doing, I just converted a deprecated `q()` call to avoid having to require `include/dba.php` anymore.

Reply:
@nupplaphil This is one of these calls that can't easily be replaced, due to the added SQL commands in the ```$sql_extra``` and ```$sql_extra4``` variables.

Reply:
I think he meant moving the query as is in a new `Model\Item` method.